<?xml version="1.0" encoding="utf-8"?>
<launch>
    <arg name="veh" default="$(env VEHICLE_NAME)" />
    <arg name="config" default="default" />
    <arg name="param_file_name" default="default" />
    
    <!-- Mock camera parameters -->
    <arg name="publish_rate" default="10.0" doc="Camera frame rate in Hz" />
    <arg name="image_width" default="640" doc="Image width in pixels" />
    <arg name="image_height" default="480" doc="Image height in pixels" />
    <arg name="mock_type" default="synthetic" doc="Type of mock image: synthetic, static, or noise" />
    
    <group ns="$(arg veh)">
        <node name="camera_node" pkg="camera_mock" type="camera_mock_node.py" output="screen">
            <param name="publish_rate" value="$(arg publish_rate)" />
            <param name="image_width" value="$(arg image_width)" />
            <param name="image_height" value="$(arg image_height)" />
            <param name="mock_type" value="$(arg mock_type)" />
        </node>
    </group>
</launch>